lib/config/hse_gparams.c

```diff
@@ -400,6 +400,30 @@ hse_gparams_get(
     return param_get(&p, pspecs, NELEM(pspecs), param, buf, buf_sz, needed_sz);
 }
 
+merr_t
+hse_gparams_set(
+    const struct hse_gparams *const params,
+    const char *const               param,
+    const char *const               value)
+{
+    merr_t                 err;
+    int                    rc;
+    static pthread_mutex_t lock = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
Pattern that needs thinking about. Although I don't anticipate this and similar function being used frequently, the mutexes are not in extremely logical places. Ideally they would be attached to the params struct itself, but copying mutexes as is done in *_params_defaults() and in ikvdb.c could lead to undefined behavior if my research is correct. Happy to be corrected.

Reply:
Sounds good.